src/bundle.c

```diff
@@ -179,15 +183,18 @@ static void required_by(struct list **reqd_by, const char *bundle_name, struct m
 
 			if (strcmp(name, bundle_name) == 0) {
 				char *bundle_str = NULL;
+
 				indent = (recursion - 1) * 4;
-				if (recursion == 1) {
+				if (!tree_view) {
+					string_or_die(&bundle_str, "%s\n", bundle->component);
```

Comment:
Instead of storing info to be printed later is way better to print what we need to print here so we don't need to keep everything in memory to print later. I'd try to refactor the calls of this function to prevent that.